### ISSUE TITLE
Xclim package not loading

### ISSUE THREAD

**Issue by kushankb** [microsoft/PlanetaryComputer#212]
Hi there, 
I am a recent planetary computer convert. Till last week, I was able to load the xclim package on the hub with no issue. Today, when I am trying to load the xclim package, I get the following error: ModuleNotFoundError: No module named 'pint.unit'. It looks like there is an issue with the pint package dependency. Is there any fix? 

Thanks!

**Comment by TomAugspurger** [comment]
Thanks, I'm working on that currently.

tl/dr, we're running into https://github.com/Ouranosinc/xclim/issues/1216. I'm going to fix it temporarily by pinning our version of `pint` to 0.19.2, and will look into moving to a newer version of xclim later (for some reason the environment isn't solving with the latest version).

I'll have a PR to https://github.com/microsoft/planetary-computer-containers shortly, and it'll be deployed later today.

You can work around it temporarily by `pip install pint==0.19.2` and restarting your Python kernel.

**Comment by kushankb** [comment]
I faced the same issue on my local machine as well. I guess a lot more people will be facing a similar issue. Thanks for the quick response! Appreciate it.

On May 4, 2023, at 10:27 AM, Tom Augspurger ***@***.******@***.***>> wrote:



Thanks, I'm working on that currently.

tl/dr, we're running into Ouranosinc/xclim#1216<https://github.com/Ouranosinc/xclim/issues/1216>. I'm going to fix it temporarily by pinning our version of pint to 0.19.2, and will look into moving to a newer version of xclim later (for some reason the environment isn't solving with the latest version).

I'll have a PR to https://github.com/microsoft/planetary-computer-containers shortly, and it'll be deployed later today.

You can work around it temporarily by pip install pint==0.19.2 and restarting your Python kernel.

**Comment by kushankb** [comment]
Installing pint ==0.19.2 does not completely resolve the issue. I am able to load the xclim package but, its creating additional errors while trying to run some of the xclim indices (eg. frost_days = xclim.indicators.atmos.frost_days(ds.tasmin, thresh ='0 degC', freq = "YS") ).


On May 4, 2023, at 10:29 AM, kushank bajaj ***@***.******@***.***>> wrote:

I faced the same issue on my local machine as well. I guess a lot more people will be facing a similar issue. Thanks for the quick response! Appreciate it.

On May 4, 2023, at 10:27 AM, Tom Augspurger ***@***.******@***.***>> wrote:



Thanks, I'm working on that currently.

tl/dr, we're running into Ouranosinc/xclim#1216<https://github.com/Ouranosinc/xclim/issues/1216>. I'm going to fix it temporarily by pinning our version of pint to 0.19.2, and will look into moving to a newer version of xclim later (for some reason the environment isn't solving with the latest version).

I'll have a PR to https://github.com/microsoft/planetary-computer-containers shortly, and it'll be deployed later today.

You can work around it temporarily by pip install pint==0.19.2 and restarting your Python kernel.

**Comment by TomAugspurger** [comment]
Right now it looks like we're stuck with what we have. The hard pin on `pandas<2` by xclim 0.42 is incompatible with other things in our environment.

I believe I did get a basic example working with xclim 0.42, so if you're still seeing errors you might want to make a reproducible bug report and submit it to the issue tracker.

We'll upgrade the containers once xclim makes their next release.

**Comment by kushankb** [comment]
Thanks for your help, Tom! Installing xclim 0.42 seem to be working on a basic example for me too. Hope nothing changes. I’ll submit a reproducible bug report, if anything changes.




On May 5, 2023, at 4:44 AM, Tom Augspurger ***@***.******@***.***>> wrote:



Right now it looks like we're stuck with what we have. The hard pin on pandas<2 by xclim 0.42 is incompatible with other things in our environment.

I believe I did get a basic example working with xclim 0.42, so if you're still seeing errors you might want to make a reproducible bug report and submit it to the issue tracker.

We'll upgrade the containers once xclim makes their next release.

**Comment by TomAugspurger** [comment]
I tried to update this last week, but ran into an issue with conda-lock. I'll try again this week.

**Comment by TomAugspurger** [comment]
This is all fixed now. Closing.
